dask/bag/core.py

```diff
@@ -244,20 +280,39 @@ def __init__(self, dsk, name, npartitions):
         self.npartitions = npartitions
         self.str = StringAccessor(self)
 
-    def map(self, func):
+    def map(self, func, **kwargs):
         """ Map a function across all elements in collection
 
         >>> import dask.bag as db
         >>> b = db.from_sequence(range(5))
         >>> list(b.map(lambda x: x * 10))  # doctest: +SKIP
         [0, 10, 20, 30, 40]
+
+        Any **kwargs are passed through to func along with the (maybe unpacked)
+        item argument(s).  Values in **kwargs must either be a non-dask python
+        literal or a dask.bag.Item.
+
+        If a dask.Bag.Item is used its computed value is passed to the map
+        function.  This means that whatever computations are implied by any
+        Item values block any of the map work.  Example:
+
+        >>> b = db.from_sequence(range(5))
+        >>> b.map(
+        ...     lambda x, total=0: float(x) / total,
+        ...     total=b.sum()).sum().compute()
+        1.0
```

Comment:
Sure I can totally see having an example in doc string with a constant python value, but we also want one with a dask value as well, I think simply using temporary variables to make the code clearer, rather than going along with the penchant for heavy inlining that dask's source code has would help:
```python
>>> nums = db.from_sequence(...)
>>> nums_total = nums.sum()
>>> normalized = b.map(lambda x, total=0: float(x) / total, total=nums_total)
>>> normalized.sum()
1.0
```